pkg/codegen/go/gen_program_expressions.go

```diff
@@ -231,36 +288,55 @@ func (g *generator) GenObjectConsExpression(w io.Writer, expr *model.ObjectConsE
 
 func (g *generator) genObjectConsExpression(w io.Writer, expr *model.ObjectConsExpression, destType model.Type) {
 	if len(expr.Items) > 0 {
-		var temps []*ternaryTemp
+		var temps []interface{}
+		isInput := isInputty(destType)
+		typeName := argumentTypeName(expr, destType, isInput)
+		if strings.HasSuffix(typeName, "Args") {
+			isInput = true
+		}
+		// invokes are not inputty
+		if strings.Contains(typeName, ".Lookup") {
+			isInput = false
+		}
```

Comment:
`destType` is a `Union<T, Output<T>> in this case so `isInputty` returns true. I suppose that some of this could be encapsulated or made more robust, but this is a simple workaround for the time being.

Reply:
Okay. I think we ought to investigate this a bit more so we can understand where that `destType` is coming from, but that can be done as follow-up.

Reply:
https://github.com/pulumi/pulumi/issues/4767